elide-core/src/main/java/com/yahoo/elide/security/checks/ExtractedChecks.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.security.checks;
+
+import com.yahoo.elide.audit.InvalidSyntaxException;
+import com.yahoo.elide.core.PersistentResource;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
+
+/**
+ * Extracted checks.
+ */
+@Slf4j
+public class ExtractedChecks {
+    private final Class<? extends Check>[] anyEntityChecks;
+    private final Class<? extends Check>[] allEntityChecks;
+    private final Class<? extends Check>[] anyFieldChecks;
+    private final Class<? extends Check>[] allFieldChecks;
+
+    /**
+     * Constructor.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation class
+     * @param <A> Type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final PersistentResource resource, final Class<A> annotationClass) {
+        this(resource, annotationClass, null);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param resource Resource
+     * @param annotationClass Annotation class
+     * @param field Field
+     * @param <A> Type parameter
+     */
+    public <A extends Annotation> ExtractedChecks(final PersistentResource resource,
```

Comment:
I think this might be cleaner if we break this up a bit.  I would have a class that includes the check(s) plus mode and maybe add static methods to build one of those classes given an annotation and an entity OR an annotation, entity,and field.